File: dev/assets/storybook.js
// Dev/test-only storybook JS (served straight from source by Dev.Endpoint, no
// bundler). Loaded as an ES module so it can import the published EexCharts
// hook verbatim plus the Phoenix/LiveView ESM builds, then hand them to
// phoenix_storybook via `window.storybook`. This is what lets the catalog's
// LiveSocket connect and drive hover tooltips / legend toggles.
import { Socket } from "/vendor/phoenix/phoenix.mjs";
import { LiveSocket } from "/vendor/live_view/phoenix_live_view.esm.js";
import EexCharts from "/eexcharts/eexcharts.js";

window.storybook = {
  Hooks: { EexCharts },
  Phoenix: { Socket },
  LiveView: { LiveSocket },
};

File: dev/storybook.ex
defmodule Dev.Storybook do
  @moduledoc """
  Dev/test-only phoenix_storybook backend for the eexcharts catalog.

  Stories live under `dev/storybook/`; the custom JS/CSS (hook registration +
  eexcharts stylesheet) are served from `dev/assets/` by `Dev.Endpoint`. The JS
  is an ES module so it can `import` the library hook and the Phoenix/LiveView
  ESM builds without a bundler.
  """
  use PhoenixStorybook,
    otp_app: :eexcharts,
    content_path: Path.expand("storybook", __DIR__),
    css_path: "/assets/storybook.css",
    js_path: "/assets/storybook.js",
    js_script_type: "module"
end
